### PR TITLE
avoid multiple zookeeper connection in consumer service

### PR DESCRIPTION
**Pull request by rohitvyavahare**

### CODE REVIEW

File: src/main/java/com/linkedin/kmf/services/ConsumeService.java [hunk 1]
```diff
@@ -238,6 +241,11 @@ public synchronized void stop() {
 
   @Override
   public void awaitShutdown() {
+    try {
+      _handlePartitionLeaderInfoExecutor.awaitTermination(10000, TimeUnit.MILLISECONDS);
```
Comment (mpetazzoni) [inline src/main/java/com/linkedin/kmf/services/ConsumeService.java]: Why not just `10, TimeUnit.SECONDS` ?

File: src/main/java/com/linkedin/kmf/services/ConsumeService.java [hunk 2]
```diff
@@ -238,6 +241,11 @@ public synchronized void stop() {
...
+    } catch (InterruptedException e) {
+      LOG.warn(_name + "/ConsumeService while trying to close _handlePartitionLeaderInfoExecutor.", e);
```
Comment (mpetazzoni) [inline src/main/java/com/linkedin/kmf/services/ConsumeService.java]: Fix message for grammar

```java
LOG.warn(_name + "/ConsumeService could not stop partition leader info executor in time", e);
```

**Review by mpetazzoni — CHANGES_REQUESTED** [review]

File: src/main/java/com/linkedin/kmf/services/ConsumeService.java (hunk 2 above)
Comment (rohitvyavahare) [reply src/main/java/com/linkedin/kmf/services/ConsumeService.java]: @mpetazzoni  make sense.. done

File: src/main/java/com/linkedin/kmf/services/ConsumeService.java (hunk 1 above)
Comment (rohitvyavahare) [reply src/main/java/com/linkedin/kmf/services/ConsumeService.java]: @mpetazzoni ummm.. ok

**Review by mpetazzoni — APPROVED** [review]

**Pull request merged**